ZenPacks/zenoss/OpenStackInfrastructure/modeler/plugins/zenoss/OpenStackInfrastructure.py

```diff
@@ -596,10 +698,10 @@ def process(self, device, results, unused):
         for hostname in sorted(nova_api_hosts):
             title = '{0}@{1} ({2})'.format('nova-api', hostname, device.zOpenStackRegionName)
             host_id = prepId("host-{0}".format(hostname))
-            nova_api_id = prepId('service-nova-api-{0}-{1}'.format(service.get('host', ''), device.zOpenStackRegionName))
+            nova_api_id = prepId('service-nova-api-{0}-{1}'.format(hostname, device.zOpenStackRegionName))
 
             services.append(ObjectMap(
-                modname='ZenPacks.zenoss.OpenStackInfrastructure.NovaApi',
+                modname='ZenPacks.zenoss.OpenStackInfrastructure.NovaService',
```

Comment:
I think it might be possible to do, but i don't think it's this easy.  More investigation is required.